polaris.shopify.com/src/components/MobileNav/MobileNav.tsx

```diff
@@ -0,0 +1,247 @@
+import Button from "../Button";
+import type { NavItem } from "../Nav";
+
+import shopifyLogo from "../../../public/shopify-logo.svg";
+import styles from "./MobileNav.module.scss";
+
+const componentsNavItems = getComponentNav();
+
+const navItems: NavItem[] = [
+  {
+    title: "Getting started",
+    url: "/",
+  },
+  ...foundationsNavItems,
+  {
+    title: "Components",
+    url: "/components",
+    children: componentsNavItems[0].children,
+  },
+  {
+    title: "Tokens",
+    url: "/tokens/colors",
+  },
+  {
+    title: "Icons",
+    url: "/icons",
+  },
+  // ...contributingNavItems,
+];
+
+interface Props {
+  currentPath: string;
+}
+
+function MobileNav({ currentPath }: Props) {
+  const [showMenu, setShowMenu] = useState(false);
+  const menuButtonRef = useRef<HTMLButtonElement>(null);
+  const logoRef = useRef<HTMLAnchorElement>(null);
+  const closeButtonRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    function hideSideNavOnResize() {
+      if (window.innerWidth > Breakpoints.Desktop && showMenu) {
+        setShowMenu(false);
+      }
+    }
+
+    window.addEventListener("resize", hideSideNavOnResize);
+
+    return () => window.removeEventListener("resize", hideSideNavOnResize);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const handleOnKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setShowMenu(false);
+      }
+    };
+
+    if (showMenu) {
+      document.addEventListener("keydown", handleOnKeyDown);
+      logoRef.current instanceof HTMLElement && logoRef.current.focus();
+    }
+
+    return () => document.removeEventListener("keydown", handleOnKeyDown);
+  }, [showMenu]);
+
+  const handleCloseMenu = () => {
+    setShowMenu(false);
+    menuButtonRef.current?.focus();
+  };
+
+  return (
+    <>
+      <Button
+        id="menu-button"
+        aria-label="Open menu"
+        aria-controls="side-menu"
+        aria-expanded={showMenu}
+        onClick={() => setShowMenu(true)}
+        ref={menuButtonRef}
+      >
+        <svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+          <path d="M19 11h-18a1 1 0 0 1 0-2h18a1 1 0 1 1 0 2zm0-7h-18a1 1 0 0 1 0-2h18a1 1 0 1 1 0 2zm0 14h-18a1 1 0 0 1 0-2h18a1 1 0 0 1 0 2z" />
+        </svg>
+      </Button>
+
+      <nav
+        id="side-menu"
+        className={className(styles.MobileNav, showMenu && styles.show)}
+      >
+        <ul className={styles.LinksList}>
+          <li>
+            <Link href="/" passHref>
+              <a
+                ref={logoRef}
+                className={styles.Logo}
+                onClick={handleCloseMenu}
+                onKeyDown={(e) => {
+                  if (e.key === "Tab" && e.shiftKey) {
+                    e.preventDefault();
+                    const closeButton = closeButtonRef.current;
+                    closeButton instanceof HTMLElement && closeButton.focus();
+                  }
```

Comment:
Room for debate, but I can see this being extracted into a function above the return. That way, the JSX part of the component becomes easy to read which is inviting to designer/developer types.

Reply:
Yeah!